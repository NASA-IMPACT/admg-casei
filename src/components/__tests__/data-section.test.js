import React from "react"
import { create } from "react-test-renderer"

import DataSection from "../data-section"
import { instrumentQuery } from "../../../test/__fixtures__"

describe("Data Section", () => {
  it("matches snapshot", () => {
    const tree = create(
      <DataSection
        id="data"
        dois={instrumentQuery.data.instrument.dois}
        filterBy={["campaigns", "platforms", "formats"]}
        category="instrument"
      />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
