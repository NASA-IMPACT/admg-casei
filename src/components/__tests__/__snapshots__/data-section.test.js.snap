// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Section matches snapshot 1`] = `
<section
  className="section__Container-sc-1oozlr5-0 bIfvkW"
  data-cy="data-section"
  mode="darkTheme"
>
  <div
    className="section__Buffer-sc-1oozlr5-1 eTpLgr"
    id="data"
  />
  <div
    className="section-header___StyledDiv-sc-1i4pe2i-0 SXaMi"
    data-cy="data-section-header"
  >
    <a
      href="#data"
    >
      <h2>
        Data Products
      </h2>
    </a>
  </div>
  <div
    className="section__SectionContent-sc-1oozlr5-2 ivTKjv"
    mode="darkTheme"
  >
    <p>
      Filter data products from this 
      instrument
       by specific
       
      campaigns
      , 
      platforms
      , or 
      formats
      .
    </p>
    <div
      className="data-section___StyledDiv-sc-11szotn-0 qHvyW"
    >
      <div
        className="filter-box___StyledDiv-sc-1n4l7nk-0 koGaSP"
      >
        <span
          id="Campaigns"
          style={
            Object {
              "border": 0,
              "clip": "rect(0 0 0 0)",
              "height": "1px",
              "margin": "-1px",
              "overflow": "hidden",
              "padding": 0,
              "position": "absolute",
              "whiteSpace": "nowrap",
              "width": "1px",
              "wordWrap": "normal",
            }
          }
        >
          Campaigns
        </span>
        <div
          data-cy="filter-by-Campaigns"
          data-reach-listbox-input=""
          data-state="closed"
          data-value={null}
          id="Campaigns-listbox"
        >
          <span
            aria-haspopup="listbox"
            aria-labelledby="Campaigns button--Campaigns-listbox"
            className="filter-box___StyledListboxButton-sc-1n4l7nk-1 hmgQBt"
            data-reach-listbox-button=""
            id="button--Campaigns-listbox"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            role="button"
            tabIndex={0}
          >
            CAMPAIGNS
            <span
              aria-hidden={true}
              data-reach-listbox-arrow=""
            >
              ▼
            </span>
          </span>
        </div>
      </div>
      <div
        className="filter-box___StyledDiv-sc-1n4l7nk-0 koGaSP"
      >
        <span
          id="Platforms"
          style={
            Object {
              "border": 0,
              "clip": "rect(0 0 0 0)",
              "height": "1px",
              "margin": "-1px",
              "overflow": "hidden",
              "padding": 0,
              "position": "absolute",
              "whiteSpace": "nowrap",
              "width": "1px",
              "wordWrap": "normal",
            }
          }
        >
          Platforms
        </span>
        <div
          data-cy="filter-by-Platforms"
          data-reach-listbox-input=""
          data-state="closed"
          data-value={null}
          id="Platforms-listbox"
        >
          <span
            aria-haspopup="listbox"
            aria-labelledby="Platforms button--Platforms-listbox"
            className="filter-box___StyledListboxButton-sc-1n4l7nk-1 hmgQBt"
            data-reach-listbox-button=""
            id="button--Platforms-listbox"
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            role="button"
            tabIndex={0}
          >
            PLATFORMS
            <span
              aria-hidden={true}
              data-reach-listbox-arrow=""
            >
              ▼
            </span>
          </span>
        </div>
      </div>
    </div>
    <div
      className="data-section___StyledDiv2-sc-11szotn-1 cOIOPW"
    >
      <div
        className="data-section___StyledDiv3-sc-11szotn-2 KRmSO"
        data-cy="data-product"
      >
        <div>
          <label
            className="label___StyledLabel-sc-b3994k-0 izJhRE"
            data-cy="doi-label"
          >
            GPM GROUND VALIDATION ADVANCED MICROWAVE PRECIPITATION RADIOMETER (AMPR) IPHEX V2
          </label>
          
        </div>
        <div
          className="data-section___StyledDiv4-sc-11szotn-3 hZXqEX"
        >
          <div
            data-cy="data-product-campaigns"
          >
            <label
              className="label___StyledLabel-sc-b3994k-0 JLqlD"
              data-cy="doi-campaign-label"
            >
              Campaigns
            </label>
            <a
              href="/campaign/IPHEx"
            >
              <small>
                Integrated Precipitation and Hydrology Experiment
              </small>
            </a>
          </div>
          <div
            data-cy="data-product-platforms"
          >
            <label
              className="label___StyledLabel-sc-b3994k-0 JLqlD"
              data-cy="doi-platform-label"
            >
              Platforms
            </label>
            <a
              href="/platform/ER-2"
            >
              <small>
                NASA Earth Resources-2 aircraft
              </small>
            </a>
          </div>
        </div>
      </div>
      <div
        className="data-section___StyledDiv3-sc-11szotn-2 KRmSO"
        data-cy="data-product"
      >
        <div>
          <label
            className="label___StyledLabel-sc-b3994k-0 izJhRE"
            data-cy="doi-label"
          >
            CAMP2Ex P-3 Remotely Sensed Aerosol Data
          </label>
          <a
            className="external-link___StyledA-sc-3ezy1s-0 iQSFnp"
            data-cy="doi-link"
            href="http://dx.doi.org/10.5067/Airborne/CAMP2Ex_Aerosol_AircraftRemoteSensing_P3_Data_1"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              height="16"
              role="img"
              version="1.1"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                External Link
              </title>
              <rect
                fill="none"
                height="16"
                width="16"
              />
              <path
                d="M3,5h4V3H1v12h12V9h-2v4H3V5z M16,8V0L8,0v2h4.587L6.294,8.294l1.413,1.413L14,3.413V8H16z"
                fill="#aac9ff"
              />
            </svg>
            <span>
              10.5067/Airborne/CAMP2Ex_Aerosol_AircraftRemoteSensing_P3_Data_1
            </span>
          </a>
        </div>
        <div
          className="data-section___StyledDiv4-sc-11szotn-3 hZXqEX"
        >
          <div
            data-cy="data-product-campaigns"
          >
            <label
              className="label___StyledLabel-sc-b3994k-0 JLqlD"
              data-cy="doi-campaign-label"
            >
              Campaigns
            </label>
            <a
              href="/campaign/CAMP2Ex"
            >
              <small>
                Clouds, Aerosol, and Monsoon Processes-Philippines Experiment
              </small>
            </a>
          </div>
          <div
            data-cy="data-product-platforms"
          >
            <label
              className="label___StyledLabel-sc-b3994k-0 JLqlD"
              data-cy="doi-platform-label"
            >
              Platforms
            </label>
            <a
              href="/platform/P-3"
            >
              <small>
                P-3 Orion
              </small>
            </a>
          </div>
        </div>
      </div>
      <div
        className="data-section___StyledDiv3-sc-11szotn-2 KRmSO"
        data-cy="data-product"
      >
        <div>
          <label
            className="label___StyledLabel-sc-b3994k-0 izJhRE"
            data-cy="doi-label"
          >
            GPM GROUND VALIDATION ADVANCED MICROWAVE PRECIPITATION RADIOMETER (AMPR) MC3E V1
          </label>
          
        </div>
        <div
          className="data-section___StyledDiv4-sc-11szotn-3 hZXqEX"
        >
          <div
            data-cy="data-product-campaigns"
          >
            <label
              className="label___StyledLabel-sc-b3994k-0 JLqlD"
              data-cy="doi-campaign-label"
            >
              Campaigns
            </label>
            <a
              href="/campaign/MC3E"
            >
              <small>
                Mid-latitude Continental Convective Clouds Experiment
              </small>
            </a>
          </div>
          <div
            data-cy="data-product-platforms"
          >
            <label
              className="label___StyledLabel-sc-b3994k-0 JLqlD"
              data-cy="doi-platform-label"
            >
              Platforms
            </label>
            <a
              href="/platform/ER-2"
            >
              <small>
                NASA Earth Resources-2 aircraft
              </small>
            </a>
          </div>
        </div>
      </div>
      <div
        className="data-section___StyledDiv3-sc-11szotn-2 KRmSO"
        data-cy="data-product"
      >
        <div>
          <label
            className="label___StyledLabel-sc-b3994k-0 izJhRE"
            data-cy="doi-label"
          >
            GPM Ground Validation Advanced Microwave Precipitation Radiometer (AMPR) OLYMPEX V1
          </label>
          
        </div>
        <div
          className="data-section___StyledDiv4-sc-11szotn-3 hZXqEX"
        >
          <div
            data-cy="data-product-campaigns"
          >
            <label
              className="label___StyledLabel-sc-b3994k-0 JLqlD"
              data-cy="doi-campaign-label"
            >
              Campaigns
            </label>
            <a
              href="/campaign/OLYMPEX"
            >
              <small>
                Olympic Mountains Experiment
              </small>
            </a>
          </div>
          <div
            data-cy="data-product-platforms"
          >
            <label
              className="label___StyledLabel-sc-b3994k-0 JLqlD"
              data-cy="doi-platform-label"
            >
              Platforms
            </label>
            <a
              href="/platform/ER-2"
            >
              <small>
                NASA Earth Resources-2 aircraft
              </small>
            </a>
          </div>
        </div>
      </div>
      <div
        className="data-section___StyledDiv3-sc-11szotn-2 KRmSO"
        data-cy="data-product"
      >
        <div>
          <label
            className="label___StyledLabel-sc-b3994k-0 izJhRE"
            data-cy="doi-label"
          >
            TCSP AMPR BRIGHTNESS TEMPERATURE (TB) V2
          </label>
          
        </div>
        <div
          className="data-section___StyledDiv4-sc-11szotn-3 hZXqEX"
        >
          <div
            data-cy="data-product-campaigns"
          >
            <label
              className="label___StyledLabel-sc-b3994k-0 JLqlD"
              data-cy="doi-campaign-label"
            >
              Campaigns
            </label>
            <a
              href="/campaign/TCSP"
            >
              <small>
                Tropical Cloud Systems and Processes
              </small>
            </a>
          </div>
          <div
            data-cy="data-product-platforms"
          >
            <label
              className="label___StyledLabel-sc-b3994k-0 JLqlD"
              data-cy="doi-platform-label"
            >
              Platforms
            </label>
            <a
              href="/platform/ER-2"
            >
              <small>
                NASA Earth Resources-2 aircraft
              </small>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
